plugins/action_generate_review_assignment_records/src/generateReviewAssignments.ts

```diff
@@ -1,24 +1,30 @@
-import { Reviewer, ReviewAssignmentObject, AssignmentStatus, ApplicationData } from './types'
+import { ActionPluginInput } from '../../types'
+import { Reviewer, ReviewAssignmentObject, AssignmentStatus } from './types'
 import databaseMethods from './databaseMethods'
 
-async function generateReviewAssignments(input: any, DBConnect: any) {
+async function generateReviewAssignments({
+  parameters,
+  applicationData,
+  DBConnect,
+}: ActionPluginInput) {
   const db = databaseMethods(DBConnect)
 
+  // Get application/reviewId from applicationData if not provided in parameters
+  const applicationId = parameters?.applicationId ?? applicationData?.applicationId
+  const reviewId = parameters?.reviewId ?? applicationData?.reviewData?.reviewId
+
   console.log('Generating review assignment records...')
   try {
-    const { applicationId, reviewId } = input
     // Get template information and current stage for application
-    const applicationData: ApplicationData = await DBConnect.getApplicationData(applicationId)
-    const { templateId, stageNumber, stageId } = applicationData
+    const { templateId, stageNumber, stageId } =
+      applicationData ?? (await DBConnect.getApplicationData(applicationId))
 
-    const numReviewLevels: number =
-      (await DBConnect.getNumReviewLevels(templateId, stageNumber)) || 0
+    const numReviewLevels: number = (await DBConnect.getNumReviewLevels(stageId)) || 0
 
     let nextReviewLevel = 1
 
-    // NB: reviewId comes from record_id on TriggerPayload when triggered from review table
     // For first review assignment
-    if (!reviewId) {
+    if (!parameters?.isReview) {
```

Comment:
Maybe better to add to a local variable at the top, so it's clear where is coming from?

Also `isReview` should be get either from `parameters` or from `applicationData` since isn't required field.

Reply:
Yeah, good point, have used the same logic as the changeStatus action to infer `isReview`: f6bdc7be